### PR TITLE
Added a function to separately compute and add recommended efficiency plot data to benchmarked timing results

### PR DESCRIPTION
**Pull request by Anirban166**

### CODE REVIEW

File: R/addRecommendedEfficiency.R [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  recommendedSpeedupData <- data.table(
+    threadCount = seq(1, systemThreadCount, length.out = systemThreadCount),
+    speedup = recommendedSpeedup,
+    type = "Recommended"
+  )
+
+  speedupData <- data.table(
+    expr = rep(functions, each = systemThreadCount),
+    threadCount = rep(c(1:systemThreadCount, seq(1, systemThreadCount, length.out = systemThreadCount)), length(functions)),
```
Comment (tdhock) [inline R/addRecommendedEfficiency.R]: rep(1:systemThreadCount, each=length(functions)) instead?

Comment (tdhock) [reply R/addRecommendedEfficiency.R]: this line seems overly complicated, what is it doing?

Comment (Anirban166) [reply R/addRecommendedEfficiency.R]: You're right, this does look messy and I overlooked this earlier - Thanks!

It just creates the `threadCount` column by repeating the sequence of thread counts (1:5 for 5 threads being the max available in a user's system for e.g.) for both "Ideal" and "Recommended" types, and for all the functions (so now I changed this to be repeated that many times simply, or `times = 2 * length(functions)`)

**Review by tdhock — APPROVED** [review]
looks good

**Pull request merged**